### PR TITLE
Support array of enums

### PR DESCRIPTION
**Pull request by lstrojny**
Support array of enums, so that e.g. with ODM you can do something like this

```php
#[Field(enumType: Variant::class]
private array $variants = [];
```

### CODE REVIEW

**Comment by malarzm** [comment]
@lstrojny thanks for the PR! I took the liberty and retargeted this to 3.2.0 as this is a feature.

I took a quick look into ODM and if I'm not mistaken the changes here in `doctrine/persistence` will on its own allow combining `collection` and `hash` types with `enumType`?

**Comment by lstrojny** [comment]
That is correct. In fact I am using the feature branch with ODM and it just works

File: src/Persistence/Reflection/EnumReflectionProperty.php [hunk 1]
```diff
@@ -60,9 +63,40 @@ public function getValue($object = null)
...
+     *
+     * @return ($enum is BackedEnum ? (string|int) : (string[]|int[]))
+     */
+    private function fromEnum($enum)
+    {
+        if (is_array($enum)) {
+            /** @var string[]|int[] $values */
+            $values = array_map([self::class, 'fromEnum'], $enum);
+
+            return $values;
```
Comment (malarzm) [inline src/Persistence/Reflection/EnumReflectionProperty.php]: Let's avoid recursion in favour of something along

```suggestion
            return array_map(fn(BackedEnum $e) => $e->value, $enum);
```

Comment (malarzm) [reply src/Persistence/Reflection/EnumReflectionProperty.php]: But compatible with PHP 7.2 :(

**Review by malarzm — DISMISSED** [review]
Also please squash your commits afterwards

**Comment by malarzm** [comment]
Thanks! I've PRed https://github.com/doctrine/mongodb-odm/pull/2532 with a test for future us. I'll see if there are any plans for releasing 3.2.0 here.

File: src/Persistence/Reflection/EnumReflectionProperty.php (hunk 1 above)
Comment (lstrojny) [reply src/Persistence/Reflection/EnumReflectionProperty.php]: Done

**Review by malarzm — APPROVED** [review]

**Comment by malarzm** [comment]
Thanks @lstrojny! I'll be releasing 3.2.0 shortly so you may get back to stable branch ;)

**Pull request merged**